src/cli/proto_resolver.rs

```diff
@@ -0,0 +1,59 @@
+use std::path::PathBuf;
+use std::sync::Arc;
+
+use anyhow::Context;
+use protox::file::{FileResolver, GoogleFileResolver};
+use url::Url;
+
+use crate::{FileIO, HttpIO, ProtoPathResolver};
+
+#[derive(Clone)]
+pub struct NativeProtoPathResolver;
+
+impl NativeProtoPathResolver {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+#[async_trait::async_trait]
+impl ProtoPathResolver for NativeProtoPathResolver {
+    async fn resolve<'a>(
+        &'a self,
+        path: &'a str,
+        http_io: Arc<dyn HttpIO>,
+        file_io: Arc<dyn FileIO>,
+    ) -> anyhow::Result<String> {
+        if let Ok(file) = GoogleFileResolver::new().open_file(path) {
```

Comment:
Try to compile GooggleFileResolver for WASM. It should work like native since the file IO is only performed at compile time.